File: app/src/main/java/tpcreative/co/qrscanner/common/api/response/BaseResponse.java
package tpcreative.co.qrscanner.common.api.response;
import com.google.gson.Gson;
import java.io.Serializable;

import tpcreative.co.qrscanner.model.Version;

public class BaseResponse implements Serializable {
    public String message ;
    public boolean error ;
    public String nextPage;
    public Version version;
    public String toFormResponse() {
        return new Gson().toJson(this);
    }
}

File: app/src/main/java/tpcreative/co/qrscanner/common/api/response/BaseResponse.kt
package tpcreative.co.qrscanner.common.api.response

import com.google.gson.Gson
import tpcreative.co.qrscanner.model.Version
import java.io.Serializable

open class BaseResponse : Serializable {
    var message: String? = null
    var error = false
    var nextPage: String? = null
    var qrscannerFreeRelease: Version? = null
    var qrscannerFreeInnovation: Version? = null
    var superQRScannerFreeInnovation: Version? = null
    fun toFormResponse(): String? {
        return Gson().toJson(this)
    }
}